sdk/resourcemanager/azure-resourcemanager-monitor/src/main/java/com/azure/resourcemanager/monitor/implementation/DiagnosticSettingsImpl.java

```diff
@@ -153,7 +156,14 @@ public Mono<Void> deleteByIdAsync(String id) {
 
     @Override
     public Flux<String> deleteByIdsAsync(Collection<String> ids) {
-        return BatchDeletionImpl.deleteByIdsAsync(ids, (rgName, name) -> this.inner().deleteAsync(rgName, name));
+        if (CoreUtils.isNullOrEmpty(ids)) {
+            return Flux.empty();
+        }
+        return Flux.fromIterable(ids)
+            .flatMapDelayError(id ->
+                deleteAsync(getResourceIdFromSettingsId(id), getNameFromSettingsId(id)).then(Mono.just(id)), 32, 32)
+            .onErrorMap(AggregatedManagementException::convertToManagementException)
+            .subscribeOn(ResourceManagerUtils.InternalRuntimeContext.getReactorScheduler());
```

Comment:
`this.inner.deleteAsync` actually takes `resourceUri` and `name` as parameters. There's no delete methods that take `resourceGroupName` as required by `BatchDeletionImpl`.